src/api/block.rs

```diff
@@ -67,460 +66,477 @@ impl BlockApi {
         &self,
         req: BlockTransactionRequest,
     ) -> ApiResult<BlockTransactionResponse> {
-        let block_hash = block_hash_from_string(req.block_identifier.hash.as_str())?;
-        let block_summary = self
-            .query_helper
-            .query_block_summary_by_hash(&block_hash) // TODO should probably use the "raw" variant
-            .await ?;
-        match block_summary
-            .transaction_summaries()
-            .iter()
-            .find(|t| t.hash.to_string() == req.transaction_identifier.hash)
-        {
-            None => Err(ApiError::NoTransactionsMatched),
-            Some(transaction) => Ok(BlockTransactionResponse::new(map_transaction(transaction))),
+        let tx_status =
+            self.query_helper.query_transaction_status(req.transaction_identifier.hash).await?;
+        match tx_status {
+            TransactionStatus::Finalized(finalized_status) => {
+                let tx = finalized_status
+                    .iter()
+                    .next()
+                    .ok_or_else(|| {
+                        ApiError::InternalServerError(anyhow::anyhow!(
+                            "Claimed finalized block is empty."
```

Comment:
The `finalized_status` maps block hashes to outcomes. The block hash is ignored here, if it were not then it could be compared.

So 
```
    pub async fn block_transaction(
        &self,
        req: BlockTransactionRequest,
    ) -> ApiResult<BlockTransactionResponse> {
        let tx_status =
            self.query_helper.query_transaction_status(req.transaction_identifier.hash).await?;
        if let Some((bh, tx)) = tx_status.is_finalized() {
            todo!(check that bh is the right block identifier)
            Ok(BlockTransactionResponse::new(map_transaction(tx)))
        } else {        
            Err(ApiError::NoTransactionsMatched)
        }
    }
```